kedro-telemetry/kedro_telemetry/plugin.py

```diff
@@ -49,15 +52,38 @@ def _hash(string: str) -> str:
     return hashlib.sha512(bytes(string, encoding="utf8")).hexdigest()
 
 
-def _get_hashed_username():
+def _get_or_create_uuid():
+    """
+    Reads a UUID from a configuration file or generates and saves a new one if not present.
+    """
+    config_path = user_config_dir("kedro")
+    full_path = os.path.join(config_path, CONFIG_FILENAME)
+    config = ConfigParser()
+
     try:
-        username = getpass.getuser()
-        return _hash(username)
-    except Exception as exc:
-        logger.warning(
-            "Something went wrong with getting the username. Exception: %s",
-            exc,
-        )
+        if os.path.exists(full_path):
+            config.read(full_path)
+
+            if config.has_section("telemetry") and "uuid" in config["telemetry"]:
+                try:
+                    return uuid.UUID(config["telemetry"]["uuid"]).hex
+                except ValueError:
+                    pass  # Invalid UUID found, will generate a new one
+
+        # Generate a new UUID and save it to the config file
+        if not config.has_section("telemetry"):
+            config.add_section("telemetry")
+        new_uuid = uuid.uuid4().hex
+        config.set("telemetry", "uuid", new_uuid)
+
+        os.makedirs(config_path, exist_ok=True)
+        with open(full_path, "w") as configfile:
+            config.write(configfile)
+
+        return new_uuid
+
+    except Exception as e:
+        logging.error(f"Failed to get or create UUID: {e}")
```

Comment:
Can this be splitted into two function? doing both read and write in a function is a bit too much and hard to test.

For example, how are you testing for the case when config exist but `telemetry` section does not exist?